In [24]:
# import pandas as pd
import numpy as np

import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns

import numpy as np
from scipy.stats import chi2_contingency
%matplotlib inline

import warnings
warnings.filterwarnings('ignore')

df = pd.read_csv("/home/piai/preprocessed_scale.csv")
df.head()

scale steel_kind  pt_thick  pt_width  pt_length hsb  fur_heat_temp  \
0      0          T        32      3700      15100  적용           1144   
1      0          T        32      3700      15100  적용           1144   
2      0          T        33      3600      19200  적용           1129   
3      0          T        33      3600      19200  적용           1152   
4      0          T        38      3100      13300  적용           1140   

   fur_heat_time  fur_soak_temp  fur_soak_time rolling_method  rolling_temp  \
0            116           1133             59     TMCP(온도제어)           934   
1            122           1135             53     TMCP(온도제어)           937   
2            116           1121             55     TMCP(온도제어)           889   
3            125           1127             68     TMCP(온도제어)           885   
4            134           1128             48     TMCP(온도제어)           873   

   descaling_count  fur_preheat_time furnace  
0                8                84   1호기1열  
1                8                63   1호기2열  
2                8                87   2호기1열  
3                8                73   2호기2열  
4                8                64   3호기1열

카이제곱 검정

scale과 steel_kind 카이제곱 검정 

In [27]:
df_scale = df['scale']
df_scale

0      0
1      0
2      0
3      0
4      0
      ..
989    0
990    0
991    0
992    0
993    0
Name: scale, Length: 994, dtype: int64

In [11]:
df_steel_kind = df['steel_kind']
df_steel_kind

0      T
1      T
2      T
3      T
4      T
      ..
995    C
996    C
997    C
998    C
999    C
Name: steel_kind, Length: 1000, dtype: object

In [34]:
contingency_table = pd.crosstab(df_scale, df_steel_kind)

chi2, p, dof, expected = chi2_contingency(contingency_table)

# 결과 출력
print("Chi-squared Statistic:", chi2)
print("p-value:", p)
print("Degrees of Freedom:", dof)
print("Expected Frequencies:\n", expected)

Chi-squared Statistic: 78.89479838930058
p-value: 6.550700807797746e-19
Degrees of Freedom: 1
Expected Frequencies:
 [[515.4084507 168.5915493]
 [233.5915493  76.4084507]]


p-value가 6.55×10−19로, 일반적인 유의 수준(예: 0.05)로 0.05보다 훨씬 작음. 따라서,  df_scale과 df_steel_kind 변수 간에 통계적으로 
유의미한 관계가 있음을 의미

scale과 steel_hsb 카이제곱 검정 

In [40]:
df_steel_hsb = df['hsb']
df_steel_hsb

0      적용
1      적용
2      적용
3      적용
4      적용
       ..
989    적용
990    적용
991    적용
992    적용
993    적용
Name: hsb, Length: 994, dtype: object

In [41]:
value_counts = df['hsb'].value_counts()
print(value_counts)

적용     947
미적용     47
Name: hsb, dtype: int64


In [42]:
contingency_table = pd.crosstab(df_scale, df_steel_hsb)

chi2, p, dof, expected = chi2_contingency(contingency_table)

# 결과 출력
print("Chi-squared Statistic:", chi2)
print("p-value:", p)
print("Degrees of Freedom:", dof)
print("Expected Frequencies:\n", expected)

Chi-squared Statistic: 105.51048606504317
p-value: 9.439705302426995e-25
Degrees of Freedom: 1
Expected Frequencies:
 [[ 32.34205231 651.65794769]
 [ 14.65794769 295.34205231]]


p-value가 9.44e-25로, 유의 수준(일반적으로 0.05)보다 훨씬 작으므로, df_scale과 df_steel_hsb 변수 간에 통계적으로 유의미한 관계가 있다

scale과 furnace 카이제곱 검정 

In [44]:
value_counts = df['furnace'].value_counts()
print(value_counts)

3호기1열    176
1호기2열    174
2호기1열    171
3호기2열    162
1호기1열    158
2호기2열    153
Name: furnace, dtype: int64


In [46]:
df_steel_fur_no = df['furnace']
df_steel_fur_no

0      1호기1열
1      1호기2열
2      2호기1열
3      2호기2열
4      3호기1열
       ...  
989    1호기2열
990    1호기2열
991    2호기2열
992    2호기2열
993    3호기1열
Name: furnace, Length: 994, dtype: object

In [47]:
contingency_table = pd.crosstab(df_scale, df_steel_fur_no)

chi2, p, dof, expected = chi2_contingency(contingency_table)

# 결과 출력
print("Chi-squared Statistic:", chi2)
print("p-value:", p)
print("Degrees of Freedom:", dof)
print("Expected Frequencies:\n", expected)

Chi-squared Statistic: 7.67460098873368
p-value: 0.1751043391036752
Degrees of Freedom: 5
Expected Frequencies:
 [[108.72434608 119.73440644 117.67002012 105.28370221 121.11066398
  111.47686117]
 [ 49.27565392  54.26559356  53.32997988  47.71629779  54.88933602
   50.52313883]]


p-value가 0.175로, 일반적인 유의 수준(예: 0.05)보다 크므로, df_scale과 df_steel_hsb 변수 간에 통계적으로 유의미한 관계가 없다

scale과 rolling_method 카이제곱 검정 

In [48]:
value_counts = df['rolling_method'].value_counts()
print(value_counts)

CR(제어압연)      836
TMCP(온도제어)    158
Name: rolling_method, dtype: int64


In [49]:
df_steel_rolling_method = df['rolling_method']
df_steel_rolling_method 

0      TMCP(온도제어)
1      TMCP(온도제어)
2      TMCP(온도제어)
3      TMCP(온도제어)
4      TMCP(온도제어)
          ...    
989      CR(제어압연)
990      CR(제어압연)
991      CR(제어압연)
992      CR(제어압연)
993      CR(제어압연)
Name: rolling_method, Length: 994, dtype: object

In [50]:
contingency_table = pd.crosstab(df_scale, df_steel_rolling_method)

chi2, p, dof, expected = chi2_contingency(contingency_table)

# 결과 출력
print("Chi-squared Statistic:", chi2)
print("p-value:", p)
print("Degrees of Freedom:", dof)
print("Expected Frequencies:\n", expected)

Chi-squared Statistic: 44.88003167017692
p-value: 2.0948325380804966e-11
Degrees of Freedom: 1
Expected Frequencies:
 [[575.27565392 108.72434608]
 [260.72434608  49.27565392]]


p-value가 2.094e-11로, 일반적인 유의 수준(예: 0.05)보다 훨씬 작으므로, df_scale과 df_steel_hsb 변수 간에 통계적으로 유의미한 관계가 있다